libs/sdk-common/src/lnurl/specs/withdraw.rs

```diff
@@ -33,14 +33,15 @@ pub async fn validate_lnurl_withdraw(
 
     // Send invoice to the LNURL-w endpoint via the callback
     let callback_url = build_withdraw_callback_url(&req_data, &invoice)?;
-    let callback_res: LnUrlCallbackStatus = get_parse_and_log_response(&callback_url, false)
-        .await
-        .map_err(|e| LnUrlError::ServiceConnectivity(e.to_string()))?;
-    let withdraw_status = match callback_res {
-        LnUrlCallbackStatus::Ok => LnUrlWithdrawResult::Ok {
+    let withdraw_status = match get_parse_and_log_response(&callback_url, false).await {
+        Ok(LnUrlCallbackStatus::Ok) => LnUrlWithdrawResult::Ok {
             data: LnUrlWithdrawSuccessData { invoice },
         },
-        LnUrlCallbackStatus::ErrorStatus { data } => LnUrlWithdrawResult::ErrorStatus { data },
+        Ok(LnUrlCallbackStatus::ErrorStatus { data }) => LnUrlWithdrawResult::ErrorStatus { data },
+        Err(e) if e.to_string().contains("operation timed out") => LnUrlWithdrawResult::Timeout {
```

Comment:
I don't know if this is the best solution. It makes this conditional on the exact error string from lnbits, which can change anytime even with a simple HTTP server upgrade.

Perhaps a more robust approach is to make the REST client timeout configurable in the SDK config? Then clients that expect to interact with slower endpoints can bump it to 60s?

Or maybe we can just bump the default from 30 to 60s?

What do you guys think?

Reply:
Not against merging this, just wondering if there's a better way.

Reply:
@andrei-21 since it is lnbit specific (which is also against the spec) I think @ok300 has a point.
What do you think? bumping to 60 seconds will solve this issue?

Reply:
The timeout idea was a practical suggestion for what appears to be a temporary problem, namely that lnbits is not (yet) spec-conform in handling LNURL-withdraws.

The `is_timeout` idea makes sense.

A few thoughts on the details: The right place to handle it is probably `get_and_log_response`, which is where the reqwest error originates. This means the error it typically throws, `ServiceConnectivityError`, would need two variants, a `Generic { err }` and a `Timeout`. This would propagate all the way to `get_parse_and_log_response` and `validate_lnurl_withdraw`, where it can result in the `LnUrlWithdrawResult::Timeout` you added.

Reply:
@andrei-21 I have a general question about this.

AFAIK the timeout would not distinguish between
- a connection issue (network, etc)
- a server issue (down for maintenance, etc)
- the synchronous payment issue (lnbits trying to pay the invoice, which takes >30s, so timeout)

In the PR its written
> ... a proposal to propagate this timeout back to the client and let him decide if the request needs to be retried, tolerated, or failed.

Isn't it a bad idea for the client to retry a timed-out Withdraw, not knowing the actual root cause of the timeout? Couldn't the 2nd, 3rd Withdraw call lead to a 2nd and 3rd actual withdrawal, and not just a "retry"?

Or am I missing something and there's a way to tell the difference?